components/command/command-handler/src/main/java/uk/gov/justice/services/core/aggregate/AggregateService.java

```diff
@@ -66,7 +84,37 @@ private Object convertEnvelopeToEvent(final JsonEnvelope event) {
         if (!eventMap.containsKey(name)) {
             throw new IllegalStateException(format("No event class registered for events of type %s", name));
         }
-
         return jsonObjectToObjectConverter.convert(event.payloadAsJsonObject(), eventMap.get(name));
     }
-}
+
+    /**
+     * Sets object input stream strategy.
+     *
+     * @param objectInputStreamStrategy the object input stream strategy
+     */
+    public void setObjectInputStreamStrategy(final ObjectInputStreamStrategy objectInputStreamStrategy) {
+        this.objectInputStreamStrategy = objectInputStreamStrategy;
+    }
+
+    private <T extends Aggregate> VersionedAggregate<T> deleteAllSnapshotsAndMakeAggregateOf(final EventStream stream,
+                                                                                             final Class<T> clazz) {
+        logger.trace("Deleting existing snapshots aggregate for instance {} of aggregate type {}", stream.getId(), clazz);
+
+        snapshotService.removeAllSnapshots(stream.getId(), clazz);
+        return snapshotService.getNewVersionedAggregate(clazz);
```

Comment:
Method calls can be combined as one call inside the SnapshotService. Once combined this method can be in lined into caller.